### ISSUE TITLE
Error: list out of range converting ipynb to tex

### ISSUE THREAD

**Issue by illava** [jupyter/nbconvert#728]
The nbconvert gave an error converting the ipython notebook(01-python-tools\01.04-use-anaconda.ipynb). 

nbconvert version:
5.2.1 and 5.3.1 have same problem 

Python 3.6.1 |Anaconda custom (64-bit)| (default, May 11 2017, 13:25:24) [MSC v.1900 64 bit (AMD64)] on win32

The doc's url is:
https://github.com/lijin-THU/notes-python/blob/master/01-python-tools/01.04-use-anaconda.ipynb

Thanks any way,
illava.
```
01-python-tools\01.01-python-overview.ipynb
01-python-tools\01.02-ipython-interpreter.ipynb
01-python-tools\01.03-ipython-notebook.ipynb
01-python-tools\01.04-use-anaconda.ipynb
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-40-e0009b956453> in <module>()
      5             path=os.path.join(folder,file)
      6             print(path)
----> 7             tex=nbconvert.export(nbconvert.LatexExporter,path)[0]
      8             l=tex.find("\maketitle")+len("\maketitle")
      9             r=tex.find("% Add a bibliography block to the postdoc")

~\Anaconda3\lib\site-packages\nbconvert\exporters\base.py in export(exporter, nb, **kw)
     80         output, resources = exporter_instance.from_notebook_node(nb, resources)
     81     elif isinstance(nb, string_types):
---> 82         output, resources = exporter_instance.from_filename(nb, resources)
     83     else:
     84         output, resources = exporter_instance.from_file(nb, resources)

~\Anaconda3\lib\site-packages\nbconvert\exporters\exporter.py in from_filename(self, filename, resources, **kw)
    170 
    171         with io.open(filename, encoding='utf-8') as f:
--> 172             return self.from_file(f, resources=resources, **kw)
    173 
    174 

~\Anaconda3\lib\site-packages\nbconvert\exporters\exporter.py in from_file(self, file_stream, resources, **kw)
    188 
    189         """
--> 190         return self.from_notebook_node(nbformat.read(file_stream, as_version=4), resources=resources, **kw)
    191 
    192 

~\Anaconda3\lib\site-packages\nbconvert\exporters\latex.py in from_notebook_node(self, nb, resources, **kw)
     80         self.register_filter('highlight_code',
     81                              Highlight2Latex(pygments_lexer=lexer, parent=self))
---> 82         return super(LatexExporter, self).from_notebook_node(nb, resources, **kw)
     83 
     84     def _create_environment(self):

~\Anaconda3\lib\site-packages\nbconvert\exporters\templateexporter.py in from_notebook_node(self, nb, resources, **kw)
    281 
    282         # Top level variables are passed to the template_exporter here.
--> 283         output = self.template.render(nb=nb_copy, resources=resources)
    284         return output, resources
    285 

~\Anaconda3\lib\site-packages\jinja2\asyncsupport.py in render(self, *args, **kwargs)
     74     def render(self, *args, **kwargs):
     75         if not self.environment.is_async:
---> 76             return original_render(self, *args, **kwargs)
     77         loop = asyncio.get_event_loop()
     78         return loop.run_until_complete(self.render_async(*args, **kwargs))

~\Anaconda3\lib\site-packages\jinja2\environment.py in render(self, *args, **kwargs)
   1006         except Exception:
   1007             exc_info = sys.exc_info()
-> 1008         return self.environment.handle_exception(exc_info, True)
   1009 
   1010     def render_async(self, *args, **kwargs):

~\Anaconda3\lib\site-packages\jinja2\environment.py in handle_exception(self, exc_info, rendered, source_hint)
    778             self.exception_handler(traceback)
    779         exc_type, exc_value, tb = traceback.standard_exc_info
--> 780         reraise(exc_type, exc_value, tb)
    781 
    782     def join_path(self, template, parent):

~\Anaconda3\lib\site-packages\jinja2\_compat.py in reraise(tp, value, tb)
     35     def reraise(tp, value, tb=None):
     36         if value.__traceback__ is not tb:
---> 37             raise value.with_traceback(tb)
     38         raise value
     39 

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\article.tplx in top-level template code()
      6 
      7 % Inherit from the specified cell style.
----> 8 ((* extends cell_style *))
      9 
     10 

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\style_ipython.tplx in top-level template code()
     58     ((*- set indention =  " " * (execution_count | length + 7) -*))
     59 \begin{Verbatim}[commandchars=\\\{\}]
---> 60 ((( text | add_prompts(first='{\color{' ~ prompt_color ~ '}' ~ prompt ~ '[{\\color{' ~ prompt_color ~ '}' ~ execution_count ~ '}]:} ', cont=indention) )))
     61 \end{Verbatim}
     62 ((*- endmacro *))

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\base.tplx in top-level template code()
      4 This template does not define a docclass, the inheriting class must define this.=))
      5 
----> 6 ((*- extends 'document_contents.tplx' -*))
      7 
      8 %===============================================================================

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\document_contents.tplx in top-level template code()
     48 ((* macro draw_figure(filename) -*))
     49 ((* set filename = filename | posix_path *))
---> 50 ((*- block figure scoped -*))
     51     \begin{center}
     52     \adjustimage{max size={0.9\linewidth}{0.9\paperheight}}{((( filename )))}

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\skeleton\display_priority.tplx in top-level template code()
      3 
      4 
----> 5 ((*- extends 'null.tplx' -*))
      6 
      7 ((=display data priority=))

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\skeleton\null.tplx in top-level template code()
     28 ((*- block header -*))
     29 ((*- endblock header -*))
---> 30 ((*- block body -*))
     31 ((*- for cell in nb.cells -*))
     32     ((*- block any_cell scoped -*))

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\base.tplx in block "body"()
    171     ((* endblock predoc *))
    172 
--> 173     ((( super() )))
    174 
    175     % Add a bibliography block to the postdoc

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\skeleton\null.tplx in block "body"()
     30 ((*- block body -*))
     31 ((*- for cell in nb.cells -*))
---> 32     ((*- block any_cell scoped -*))
     33         ((*- if cell.cell_type == 'code'-*))
     34             ((*- if resources.global_content_filter.include_code -*))

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\skeleton\null.tplx in block "any_cell"()
     83         ((*- elif cell.cell_type in ['markdown'] -*))
     84             ((*- if resources.global_content_filter.include_markdown -*))
---> 85                 ((*- block markdowncell scoped-*)) ((*- endblock markdowncell -*))
     86             ((*- endif -*))
     87         ((*- elif cell.cell_type in ['raw'] -*))

~\Anaconda3\lib\site-packages\nbconvert\exporters\..\templates\latex\document_contents.tplx in block "markdowncell"()
     65 % Render markdown
     66 ((* block markdowncell scoped *))
---> 67     ((( cell.source | citation2latex | strip_files_prefix | convert_pandoc('markdown', 'json',extra_args=[]) | resolve_references | convert_pandoc('json','latex'))))
     68 ((* endblock markdowncell *))
     69 

~\Anaconda3\lib\site-packages\nbconvert\filters\filter_links.py in resolve_references(source)
     14     internally in ``pandoc``.
     15     """
---> 16     return applyJSONFilters([resolve_one_reference], source)
     17 
     18 def resolve_one_reference(key, val, fmt, meta):

~\Anaconda3\lib\site-packages\pandocfilters.py in applyJSONFilters(actions, source, format)
    195     altered = doc
    196     for action in actions:
--> 197         altered = walk(altered, action, format, meta)
    198 
    199     return json.dumps(altered)

~\Anaconda3\lib\site-packages\pandocfilters.py in walk(x, action, format, meta)
    118                     array.append(walk(res, action, format, meta))
    119             else:
--> 120                 array.append(walk(item, action, format, meta))
    121         return array
    122     elif isinstance(x, dict):

~\Anaconda3\lib\site-packages\pandocfilters.py in walk(x, action, format, meta)
    111                              item['c'] if 'c' in item else None, format, meta)
    112                 if res is None:
--> 113                     array.append(walk(item, action, format, meta))
    114                 elif isinstance(res, list):
    115                     for z in res:

~\Anaconda3\lib\site-packages\pandocfilters.py in walk(x, action, format, meta)
    122     elif isinstance(x, dict):
    123         for k in x:
--> 124             x[k] = walk(x[k], action, format, meta)
    125         return x
    126     else:

~\Anaconda3\lib\site-packages\pandocfilters.py in walk(x, action, format, meta)
    109             if isinstance(item, dict) and 't' in item:
    110                 res = action(item['t'],
--> 111                              item['c'] if 'c' in item else None, format, meta)
    112                 if res is None:
    113                     array.append(walk(item, action, format, meta))

~\Anaconda3\lib\site-packages\nbconvert\filters\filter_links.py in resolve_one_reference(key, val, fmt, meta)
     27 
     28     if key == 'Link':
---> 29         target = val[2][0]
     30         m = re.match(r'#(.+)$', target)
     31         if m:

IndexError: list index out of range
```

**Comment by takluyver** [comment]
Check your pandoc version. That code may fail if pandoc is too old.

**Comment by illava** [comment]
That's it. 
Solved by changing pandoc 1.15.0.6 to 2.0.6.
Thanks a lot.

**Comment by t-makaro** [comment]
Closing as solved.
